packages/core/core/src/Environment.js

```diff
@@ -70,6 +70,11 @@ export default class Environment implements IEnvironment {
   }
 
   merge(env: ?EnvironmentOpts) {
+    // If merging the same object, avoid copying.
```

Comment:
Good catch. This has probably been the common case?

Reply:
Yep! And it's resulted in a lot of duplicate objects everywhere